services/backend/src/main/java/com/soprasteria/g4it/backend/apifiles/business/FileSystemService.java

```
@@ -129,7 +157,24 @@ public List<FileDescriptionRest> listTemplatesFiles() throws IOException {
      * @return the list of files
      */
     public InputStream downloadFile(final String organization, final Long workspaceId, final FileFolder fileFolder, final String filename) throws IOException {
-        return fetchStorage(organization, workspaceId.toString()).readFile(fileFolder, filename);
+        FileStorage storage = fetchStorage(organization, workspaceId.toString());
+
+        // 1. Try templates root
+        try {
+            return storage.readFile(fileFolder, filename);
+        } catch (IOException ignored) {
+        }
+
+        // 2. If template file → try inside subfolder
+        if (fileFolder == FileFolder.TEMPLATES) {
+            String pathInSubfolder = "Ready for Production/" + filename;
+            try {
+                return storage.readFile(FileFolder.TEMPLATES, pathInSubfolder);
+            } catch (IOException ignored) {
```

plz add a comment that exception is ignored intentionally